### PR TITLE
Improve pom.xml

### PR DESCRIPTION
**Pull request by pan3793**
### What changes in this PR?

- correct `log4j.version` to` slf4j.version`
- extract versions to properties

And, we usually pin dependency version and exclude transitive dependencies in `<dependencyManagement>` block, but define the `scope` in specify `<dependency>` block, do you agree with that? If yes, I'd like to do it in another PR.

### CODE REVIEW

**Comment by yaooqinn** [comment]
Thanks, @pan3793  for your first PR, and cc @ulysses-you

File: pom.xml [hunk 1]
```diff
@@ -281,7 +295,7 @@
             <dependency>
                 <groupId>org.mockito</groupId>
                 <artifactId>mockito-core</artifactId>
-                <version>1.10.19</version>
+                <version>${mockito.verison}</version>
```
Comment (yaooqinn) [inline pom.xml:298]: Hi, this dependency is useless anymore, can we remove it?

File: pom.xml [hunk 2]
```diff
@@ -66,7 +66,21 @@
         <scala.binary.version>2.12</scala.binary.version>
         <maven.version>3.6.3</maven.version>
         <spark.version>3.0.1</spark.version>
-        <log4j.version>1.7.30</log4j.version>
+        <slf4j.version>1.7.30</slf4j.version>
+        <mockito.verison>1.10.19</mockito.verison>
+
+        <maven.plugin.antrun.version>3.0.0</maven.plugin.antrun.version>
```
Comment (yaooqinn) [inline pom.xml:72]: The above version change looks fine. But I am -1 with these plugin changes. Not that straightforward as we only use them once each

File: pom.xml (hunk 1 above)
Comment (pan3793) [reply pom.xml]: Sound good, removed.

File: pom.xml (hunk 2 above)
Comment (pan3793) [reply pom.xml:71]: In my view, there are 2 benefits:
1. unify both dependencies and plugins versions to one place
2. if we want to try another version of plugin, just build with `-Dmaven.plugin.xxx.version=1.2.3`

Comment (yaooqinn) [reply pom.xml:71]: thanks for clarification

**Review by yaooqinn — APPROVED** [review]

**Comment by pan3793** [comment]
@yaooqinn 
> And, we usually pin dependency version and exclude transitive dependencies in `<dependencyManagement>` block, but define the `scope` in specify `<dependency>` block, do you agree with that? If yes, I'd like to do it in another PR.

How about this?

**Comment by yaooqinn** [comment]
ok

**Comment by yaooqinn** [comment]
thanks @pan3793  merged to master and feel free to open PRs

**Pull request closed without merging**